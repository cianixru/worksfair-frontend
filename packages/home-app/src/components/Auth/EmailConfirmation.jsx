import React, { Component } from 'react';
import PropTypes from 'prop-types';

import alert from '../utils/alert';
import { sendConfirmationEmail } from '../../utils/services';

class EmailConfirmation extends Component {
  sendConfirmationEmail = async () => {
    try {
      const username = localStorage.getItem('username');
      const email = localStorage.getItem('email');
      const token = localStorage.getItem('token');
      const url = `${
        window.location.host}/dashboard/${
        username}/businesses?token=${token}`;
      if (token && email) {
        await sendConfirmationEmail(url, email);
      } else {
        alert.error('Something went wrong. Please try to log into your account.');
      }
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    const content = {
      signup: {
        icon: 'fa-paper-plane-o',
        iconColour: 'has-text-info',
        main: 'We have sent you an email',
      },
      login: {
        icon: 'fa-exclamation-triangle',
        iconColour: 'has-text-warning',
        main: 'You have not confirmed your account',
      },
    };
    const { previousLocation } = this.props;
    return (
      <div className="message-body">
        <div>
          <div>
            <span className="icon is-normal">
              <i
                className={`fa ${
                  content[previousLocation].icon} is-size-3 ${
                  content[previousLocation].iconColour} `}
                aria-hidden="true" />
            </span>
          </div>
          <p className="is-size-4">{content[previousLocation].main}</p>
          <p className="is-size-6">
            Your account needs to be confirmed for you to use worksfair. Find the confirmation email in your inbox.
          </p>
          <p className="margin-top-25">
            <button
              className="button is-info is-outlined"
              onClick={this.sendConfirmationEmail}>
              Resend Email
            </button>
          </p>
        </div>
      </div>
    );
  }
}

EmailConfirmation.propTypes = {
  location: PropTypes.object,
  previousLocation: PropTypes.string,
};

export default EmailConfirmation;
