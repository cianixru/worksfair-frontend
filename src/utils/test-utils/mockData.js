// eslint-disable-next-line import/prefer-default-export
export const user = {
  username: 'Theookafor2',
  first_name: 'eod',
  last_name: 'okafor',
  email: 'theodorehope@andela.com',
  facebook: null,
  instagram: null,
  twitter: null,
  headline: null,
  current_location: null,
  role: 'admin',
  confirmed_account: false,
  verified_account: false,
  reset_token: null,
  image_url: null,
  created_at: '2019-03-10T08:48:42.942532Z',
  modified_at: '2019-03-10T08:48:42.942544Z',
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9',
};

export const webpage = {
  owner: {
    id: 1,
    username: 'theowa',
    first_name: 'theo',
    last_name: 'IO',
    email: 'theo.iowa@yahoo.com',
    facebook: null,
    instagram: null,
    twitter: null,
    headline: null,
    current_location: null,
    role: 'admin',
    confirmed_account: false,
    verified_account: false,
    reset_token: null,
    image_url: 'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    created_at: '2019-04-25T09:07:08.209916Z',
    modified_at: '2019-04-25T09:07:08.209930Z'
  },
  title: 'Happiness',
  sub_domain_name: 'happiness',
  description: 'Happiness truly lives here. We just happy',
  keywords: 'passion, software, engineering, technology, worksfair',
  country: 'Nigeria',
  state: 'Lagos',
  city: 'Okota',
  area: 'Omagba Phase 1, Phase 2, Abatete',
  address: 'No 7 Obinna Uzor Street, Canal Estate',
  phone: null,
  email: null,
  website: null,
  featured_images: [
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png'
  ],
  is_active: true,
  logo: null,
  colour: 'is-dark',
  created_at: '2019-04-28T18:39:25.174272Z',
  updated_at: '2019-05-01T12:18:49.669633Z'
};
