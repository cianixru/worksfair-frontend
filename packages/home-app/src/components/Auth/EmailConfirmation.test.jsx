import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import EmailConfirmation from './EmailConfirmation';
import { user } from '../../utils/test-utils/mockData';

afterEach(cleanup);

const actions = {
  EmailConfirmation: jest.fn(),
};
const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({ auth: { currentUser: { user } } });

describe('EmailConfirmation.jsx', () => {
  test('renders correctly', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <EmailConfirmation previousLocation="login" />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
