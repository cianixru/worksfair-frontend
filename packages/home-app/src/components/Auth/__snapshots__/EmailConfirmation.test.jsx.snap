// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailConfirmation.jsx renders for login correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="message-body"
      >
        <div>
          <div>
            <span
              class="icon is-normal"
            >
              <i
                aria-hidden="true"
                class="fa fa-exclamation-triangle is-size-3 has-text-warning "
              />
            </span>
          </div>
          <p
            class="is-size-4"
          >
            You have not confirmed your account
          </p>
          <p
            class="is-size-6"
          >
            Your account needs to be confirmed for you to use worksfair. Find the confirmation email in your inbox.
          </p>
          <p
            class="margin-top-25"
          >
            <button
              class="button is-info is-outlined"
            >
              Resend Email
            </button>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="message-body"
    >
      <div>
        <div>
          <span
            class="icon is-normal"
          >
            <i
              aria-hidden="true"
              class="fa fa-exclamation-triangle is-size-3 has-text-warning "
            />
          </span>
        </div>
        <p
          class="is-size-4"
        >
          You have not confirmed your account
        </p>
        <p
          class="is-size-6"
        >
          Your account needs to be confirmed for you to use worksfair. Find the confirmation email in your inbox.
        </p>
        <p
          class="margin-top-25"
        >
          <button
            class="button is-info is-outlined"
          >
            Resend Email
          </button>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EmailConfirmation.jsx renders for signup correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="message-body"
      >
        <div>
          <div>
            <span
              class="icon is-normal"
            >
              <i
                aria-hidden="true"
                class="fa fa-paper-plane-o is-size-3 has-text-info "
              />
            </span>
          </div>
          <p
            class="is-size-4"
          >
            We have sent you an email
          </p>
          <p
            class="is-size-6"
          >
            Your account needs to be confirmed for you to use worksfair. Find the confirmation email in your inbox.
          </p>
          <p
            class="margin-top-25"
          >
            <button
              class="button is-info is-outlined"
            >
              Resend Email
            </button>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="message-body"
    >
      <div>
        <div>
          <span
            class="icon is-normal"
          >
            <i
              aria-hidden="true"
              class="fa fa-paper-plane-o is-size-3 has-text-info "
            />
          </span>
        </div>
        <p
          class="is-size-4"
        >
          We have sent you an email
        </p>
        <p
          class="is-size-6"
        >
          Your account needs to be confirmed for you to use worksfair. Find the confirmation email in your inbox.
        </p>
        <p
          class="margin-top-25"
        >
          <button
            class="button is-info is-outlined"
          >
            Resend Email
          </button>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
